cuda/components/atomic.cuh

```diff
@@ -39,11 +39,73 @@ namespace kernels {
 namespace cuda {
 
 
+namespace detail {
+
+
+template <typename ValueType, typename = void>
+struct atomic_add {
+    __forceinline__ __device__ void operator()(ValueType *, ValueType)
+    {
+        GKO_ASSERT(false);
+        // TODO: add proper implementation of generic atomic add
```

Comment:
The compiler build this function, so it is failed when using `static_assert.`
I keep `GKO_ASSERT` here

Reply:
That was actually a bit tricky, I did not know that it would fail right away.
I think this should work and provide a good error report:
```
// sizeof is guaranteed that the result will always be larger than zero.
static_assert(sizeof(ValueType) == 0, "This default function is not implemented, only the specializations are.");
```
From [cppreference](https://en.cppreference.com/w/cpp/language/sizeof):
> The result of sizeof is always nonzero, even if applied to an empty class type.

apparently, C++ would also allow `= delete` on the default case, however, there seems to be an issue with Mac, so I would go with the `sizeof` version.

Reply:
it works!